### ISSUE TITLE
Firefox fails in browser-to-server

### ISSUE THREAD

**Issue by MarcoPolo** [libp2p/js-libp2p-webrtc#96]
To repro, try running the browser to server example in Firefox. I get this failure:

<img width="1027" alt="Screenshot 2023-03-15 at 11 29 01 AM" src="https://user-images.githubusercontent.com/594035/225407900-94b902fd-57de-408a-ac3c-1122309d900a.png">

@ckousik can you please take a look? Thank you

**Comment by ckousik** [comment]
I'm unable to reproduce this exact failure as I can get the peerconnection to connect. What is your setup when running this test? 

Also, a failure occurs during the noise handshake because of firefox's lack of support for `RTCCertificate.getFingerprints`. This is odd since it is required by the w3c spec: https://w3c.github.io/webrtc-pc/#rtccertificate-interface
This continues to be broken in firefox as of March 16: https://wpt.fyi/results/webrtc/RTCCertificate.html?run_id=5198639255781376&run_id=5082648899223552&run_id=5198347533549568&run_id=5116569141641216

**Comment by MarcoPolo** [comment]
latest macOS (ventura 13.2.1); latest firefox (111.0). I run the commands as stated in the README for the browser-to-server example. Same setup works in Safari.

**Comment by mxinden** [comment]
> Also, a failure occurs during the noise handshake because of firefox's lack of support for `RTCCertificate.getFingerprints`. This is odd since it is required by the w3c spec: https://w3c.github.io/webrtc-pc/#rtccertificate-interface
> This continues to be broken in firefox as of March 16: https://wpt.fyi/results/webrtc/RTCCertificate.html?run_id=5198639255781376&run_id=5082648899223552&run_id=5198347533549568&run_id=5116569141641216

@ckousik can we not extract the fingerprint from the local SDP string on Firefox?

>    On Chrome _A_ can access its TLS certificate fingerprint directly via  `RTCCertificate#getFingerprints`. Firefox does not allow _A_ to do so. Browser compatibility can be found [here](https://developer.mozilla.org/en-US/docs/Web/API/RTCCertificate). In
   practice, this is not an issue since the fingerprint is embedded in the local SDP string.

https://github.com/libp2p/specs/blob/6634ca7abb2f955645243d48d1cd2fd02a8e8880/webrtc/README.md?plain=1#L332-L337

**Comment by ckousik** [comment]
> > Also, a failure occurs during the noise handshake because of firefox's lack of support for `RTCCertificate.getFingerprints`. This is odd since it is required by the w3c spec: https://w3c.github.io/webrtc-pc/#rtccertificate-interface
> > This continues to be broken in firefox as of March 16: https://wpt.fyi/results/webrtc/RTCCertificate.html?run_id=5198639255781376&run_id=5082648899223552&run_id=5198347533549568&run_id=5116569141641216
> 
> @ckousik can we not extract the fingerprint from the local SDP string on Firefox?
> 
> > On Chrome _A_ can access its TLS certificate fingerprint directly via  `RTCCertificate#getFingerprints`. Firefox does not allow _A_ to do so. Browser compatibility can be found [here](https://developer.mozilla.org/en-US/docs/Web/API/RTCCertificate). In
> > practice, this is not an issue since the fingerprint is embedded in the local SDP string.
> 
> https://github.com/libp2p/specs/blob/6634ca7abb2f955645243d48d1cd2fd02a8e8880/webrtc/README.md?plain=1#L332-L337

Thanks Max! I don't see a reason why we cannot. Let me test this.

**Comment by p-shahi** [comment]
Update: I have asked Igalia, a partner of Protocol Labs who contributes to the browser ecosystem, to see if they can add support for the fingerprint API in Firefox. They are evaluating/prioritizing and begun discussing with the Firefox team. I shared the specification with them as well.

Upon reading the WebRTC spec, I came across this in the [Connection Security](https://github.com/libp2p/specs/tree/master/webrtc#connection-security) section:

> On Chrome A can access its TLS certificate fingerprint directly via RTCCertificate#getFingerprints. Firefox does not allow A to do so. Browser compatibility can be found [here](https://developer.mozilla.org/en-US/docs/Web/API/RTCCertificate). **In practice, this is not an issue since the fingerprint is embedded in the local SDP string.**

Does this mean that we can use the fingerprint embedded in the local SDP string as a workaround to get this working in Firefox?
cc @mxinden with whom I discussed this today (he believes it is possible)
